src/main/java/org/openpnp/machine/reference/ReferencePnpJobProcessor.java

```diff
@@ -426,10 +432,26 @@ protected void doPlan() throws Exception {
 
         fireTextStatus("Planning placements.");
 
-        // Get the list of unfinished placements and sort them by part height.
-        List<JobPlacement> jobPlacements = getPendingJobPlacements().stream()
-                .sorted(Comparator.comparing(JobPlacement::getPartHeight))
-                .collect(Collectors.toList());
+        List<JobPlacement> jobPlacements;
+
+        if (this.jobOrder == ORDER_JOB_PARTHEIGHT) {
+            // Get the list of unfinished placements and sort them by part height.
+            jobPlacements = getPendingJobPlacements().stream()
+                    .sorted(Comparator.comparing(JobPlacement::getPartHeight))
+                    .collect(Collectors.toList());
+        } else {
+            if (this.jobOrder == ORDER_JOB_PART) {
+                // Get the list of unfinished placements and sort them by part.
+                jobPlacements = getPendingJobPlacements().stream()
+                        .sorted(Comparator.comparing(JobPlacement::getPartId))
+                        .collect(Collectors.toList());
+            } else {
```

Comment:
This seems to be a duplicate of the above code?

Reply:
it is like default value if string is wrong

Reply:
anyone can easy expand list of sort directions